### ISSUE TITLE
User Edit page does not populate fully from Users table

### ISSUE THREAD

**Issue by ksg1960** [meeting-room-booking-system/mrbs-code#3688]
I have created multiple detail fields for the users in the Users table.  The Add User page prepopulates the choices for these attributes by way of 
config.inc.php:$select_options['users.school'] = array('Chemistry', 'Pharmacy', 'Engineering');

the fields like users.school that are short, seem to populate from the table when opening an Edit User page.
One field is extremely long,  being about 50 string (varchar) choices also setup in config.inc.php.  It populates using the first string in the arrayed list, instead of what is saved in the Users table. 

Browser details (please complete the following information):

OS: [Windows 10]
Browser [e.g. Edge and Chrome]
Version [e.g. 125.0.2535.51]
Server details (please complete the following information):

MRBS version [1.11.5 goto time branch]
Web server: [e.g. Apache 2.4.48]
OS: [e.g. redhat 8]
PHP version: { 7.2.24]
Database used: [ MySQL 8.0.13]

**Comment by campbell-m** [comment]
Can you clarify the problem please?  Does it occur when

- the custom field is a VARCHAR(50)?
- or, there are 50 options in `$select_options`?
- or, one of the options is 50 characters long?

**Comment by ksg1960** [comment]
It is varchar(32) and there are over 50 options in the $select_options variable.


And thank you the  systemdefaults.inc.php edit worked.   this begs the question, if I want this configuration setting to be stable across upgrades would it be possible to place the line in config.inc.php?


Scott



Scott K. Smith, Director of NMR Spectroscopy
The University of Texas at Austin | Chemistry | 832-294-1928 | utexas.edu<https://www.utexas.edu/>






________________________________

**Comment by campbell-m** [comment]
> And thank you the  systemdefaults.inc.php edit worked.   this begs the question, if I want this configuration setting to be stable across upgrades would it be possible to place the line in config.inc.php?

Yes, you should never edit systemdefaults.inc.php and always place your changes in config.inc.php.  I was just pointing you to the section of systemdefaults.inc.php which gives you more details.

**Comment by ksg1960** [comment]
[like]  Smith, Scott K reacted to your message:
________________________________

**Comment by campbell-m** [comment]
> It is varchar(32) and there are over 50 options in the $select_options variable.

Thanks.  Can you describe how to reproduce the problem please?

**Comment by campbell-m** [comment]
If the value in the database table doesn't correspond exactly to one of the options in the config file then the first option will be displayed.

It's usually better to use an associative array for `$select_options`, eg

~~~php
$select_options['users.school'] = array(
  'a' => 'Chemistry',
  'b' => 'Pharmacy',
  'c' => 'Engineering'
);
~~~

because then you can change the option description without having to change the database.

**Comment by ksg1960** [comment]
Operator Error:  first letter capitalization not consistent.    the Function works as described.
